### ISSUE TITLE
[Feature]: Support for Google AdSense

### ISSUE THREAD

**Issue by dSyncro** [g1eny0ung/hugo-theme-dream#315]
### Feature request

The theme already has support for Google Analytics. I think It would be nice to also have easy integration for Google AdSense on websites in production.

### Proposed solution

I think we could easily add a simple partial loading the script in `baseof.html` pulling required ids and params from the config like already happens with Google Analytics.

### Additional context

I also think this feature should be **opt-in** and **disabled by default**. Nobody wants to promote ads spamming on websites😜.

Nevertheless I think it might be a useful feature for consumers of the theme and have its close to ethical purposes.

**Comment by g1eny0ung** [comment]
Yes, integrating AdSense looks like easily. But there's one thing I need to consider: where should I put the advertisements?

My first thought was to put it aside on the post page:

<img width="1197" alt="image" src="https://github.com/user-attachments/assets/ecef96e6-f0d5-4a99-a7c4-6ed2ffa9a282">

I don't want to put it on the home page because the masonry layout doesn't have enough space to place it.

What do you think?

**Comment by diamantelist** [comment]
In the home page, it would be fine to put advertisements just like posts... in posts page, indeed, it would be great to put advertisements on both sides

**Comment by g1eny0ung** [comment]
> In the home page, it would be fine to put advertisements just like posts... in posts page, indeed, it would be great to put advertisements on both sides

OK, but I am still concerned about whether it is a good idea to put ads on the home page. And sure, we can also put ads on both sides of the post pages.

**Comment by WinPoss** [comment]
> > In the home page, it would be fine to put advertisements just like posts... in posts page, indeed, it would be great to put advertisements on both sides
> 
> OK, but I am still concerned about whether it is a good idea to put ads on the home page. And sure, we can also put ads on both sides of the post pages.



I think it's better this way: offer both ways and then let the user choose which one to turn on or off.

**Comment by g1eny0ung** [comment]
> > > In the home page, it would be fine to put advertisements just like posts... in posts page, indeed, it would be great to put advertisements on both sides
> > 
> > 
> > OK, but I am still concerned about whether it is a good idea to put ads on the home page. And sure, we can also put ads on both sides of the post pages.
> 
> I think it's better this way: offer both ways and then let the user choose which one to turn on or off.

I've already added a slot for AdSense on the post page. To place your AdSense script, you can create an `adsense.html` under `layouts/partials/`. It will display under the TOC on the right aside.

For the home page, I've not had permission to open my AdSense, so I still try to find a way to test it.

**Comment by g1eny0ung** [comment]
> It will display under the TOC on the right aside.

Now it moves above the TOC.

**Comment by g1eny0ung** [comment]
Hi guys, please refer to this [document](https://hugo-theme-dream.g1en.site/adsense) for the newest usage.
